### PR TITLE
Raising exception when dataset ID can't be implied in set_default...

### LINKED ISSUE #475: Add GAE/GCE support for datastore.set_default_dataset and add exception on failure
See discussion in #444 for comments on exception.

Something like [`oauth2client.client._get_environment`][1] could be good to detect GAE/GCE dataset ID's but `_detect_gce_environment` should be better as discussed in google/oauth2client#93

Not blocking for M1.

[1]: https://github.com/google/oauth2client/blob/481b4b6310e5872fbd5ed6c7aa336938e564d729/oauth2client/client.py#L953

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #475.

### CODE REVIEW

File: gcloud/datastore/__init__.py [hunk 1]
```diff
@@ -92,6 +94,8 @@ def set_default_dataset_id(dataset_id=None):
 
     if dataset_id is not None:
         _implicit_environ.DATASET_ID = dataset_id
+    else:
+        raise EnvironmentError('No dataset ID could be implied.')
```
Comment (tseaver) [inline gcloud/datastore/__init__.py]: Directionality:  s/implied/inferred/

**Comment by tseaver** [comment]
Except for the "implied" vs. "inferred" bit, LGTM

**Comment by dhermes** [comment]
Folding into original commit. I apparently can't keep these two straight :)

**Comment by dhermes** [comment]
@tseaver It seems we are experiencing a coveralls outage. I'm going to merge, cover passing locally.

Any ideas?

**Pull request merged**